diff --git a/ovn/controller/ovn-controller.h b/ovn/controller/ovn-controller.h
index 078c9eabe..20eb992db 100644
--- a/ovn/controller/ovn-controller.h
+++ b/ovn/controller/ovn-controller.h
@@ -23,7 +23,7 @@
 struct ovsrec_bridge_table;
 
 /* Linux supports a maximum of 64K zones, which seems like a fine default. */
-#define MAX_CT_ZONES 65535
+#define MAX_CT_ZONES 59999
 
 /* States to move through when a new conntrack zone has been allocated. */
 enum ct_zone_pending_state {
